pkg/services/oauthserver/external_service.go

```diff
@@ -7,27 +7,11 @@ import (
 
 	"github.com/ory/fosite"
 
+	"github.com/grafana/grafana/pkg/plugins/oauth"
```

Comment:
This would make more sense if we simplify the interface for plugins (which we probably can).

Reply:
Agree with @marefr.  I would keep contracts in this package (pkg/service/oauthserver) as well, because those contracts are related to the external services registration in general.

Reply:
I can see that the changes in this file has not been reverted. The external service registration related contracts (KeyResult, ExternalServiceDTO, ExternalServiceRegistration) should be here, in this package, because those are related to the external services registration in general.